zerver/views/message_fetch.py

```diff
@@ -991,12 +991,7 @@ def get_messages_backend(
 
     assert realm is not None
 
-    if (
-        is_web_public_query
-        or realm.email_address_visibility != Realm.EMAIL_ADDRESS_VISIBILITY_EVERYONE
-    ):
-        # If email addresses are only available to administrators,
-        # clients cannot compute gravatars, so we force-set it to false.
+    if is_web_public_query:
         client_gravatar = False
```

Comment:
Do we want to add a comment here noting that `client_gravatar` here is just the user's requested value, and `finalize_payload` is responsible for sending the avatar URL or not depending on each individual sender's email_address_visibility?